opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,536 @@
+pub(crate) struct GenevaConfigClientConfig {
+    pub endpoint: String,
+    pub environment: String,
+    pub account: String,
+    pub namespace: String,
+    pub region: String,
+    pub config_major_version: u32,
+    pub auth_method: AuthMethod, // agent_identity and agent_version are hardcoded for now
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Deserialize)]
+pub(crate) struct IngestionGatewayInfo {
+    #[serde(rename = "Endpoint")]
+    pub(crate) endpoint: String,
+    #[serde(rename = "AuthToken")]
+    pub(crate) auth_token: String,
+    #[serde(rename = "AuthTokenExpiryTime")]
+    pub(crate) auth_token_expiry_time: String,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub(crate) struct MonikerInfo {
+    pub name: String,
+    pub account_group: String,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct StorageAccountKey {
+    #[serde(rename = "AccountMonikerName")]
+    account_moniker_name: String,
+    #[serde(rename = "AccountGroupName")]
+    account_group_name: String,
+    #[serde(rename = "IsPrimaryMoniker")]
+    is_primary_moniker: bool,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct GenevaResponse {
+    #[serde(rename = "IngestionGatewayInfo")]
+    ingestion_gateway_info: IngestionGatewayInfo,
+    // Make storage_account_keys optional since it might not be present in all responses
+    #[serde(rename = "StorageAccountKeys", default)]
+    storage_account_keys: Vec<StorageAccountKey>,
+    // Keep tag_id as it might be used for validation
+    #[serde(rename = "TagId", default)]
+    tag_id: Option<String>,
+}
+
+#[allow(dead_code)]
+struct CachedAuthData {
+    // Store the complete token and moniker info
+    auth_info: Option<(IngestionGatewayInfo, MonikerInfo)>,
+    // Store the endpoint from token for quick access
+    token_endpoint: Option<String>,
+    // Store expiry separately for quick access
+    token_expiry: Option<DateTime<Utc>>,
+}
+
+#[allow(dead_code)]
+pub(crate) struct GenevaConfigClient {
+    config: GenevaConfigClientConfig,
+    http_client: Client,
+    cached_data: RwLock<CachedAuthData>,
+}
+
+/// Client for interacting with the Geneva Configuration Service.
+///
+/// This client handles authentication and communication with the Geneva Config
+/// API to retrieve configuration information like ingestion endpoints.
+impl GenevaConfigClient {
+    /// Creates a new Geneva Config Client with the provided configuration.
+    ///
+    /// # Arguments
+    /// * `config` - The client configuration
+    ///
+    /// # Returns
+    /// * `Result<Self>` - A new client instance or an error
+    ///
+    /// # Errors
+    /// * `GenevaConfigClientError::Certificate` - If certificate authentication fails
+    /// * `GenevaConfigClientError::Io` - If reading certificate files fails
+    /// * `GenevaConfigClientError::Tls` - If TLS configuration fails
+    #[allow(dead_code)]
+    pub(crate) async fn new(config: GenevaConfigClientConfig) -> Result<Self> {
```

Comment:
```suggestion
    pub(crate) fn new(config: GenevaConfigClientConfig) -> Result<Self> {
```

Reply:
I think `async` would be idiomatic, and probably future proof here - specifically it performs file I/O for certificates,  and potentially would need network i/o for Managed Identity too.

Reply:
This method is not performing any async operation. It would be against the idiomatic way to mark a method async when it does not have any await call in its implementation. If in future, this method is updated to perform asynchronous I/O, we can add the `async` keyword then.

Reply:
fair enough. Have made it sync.